add select distro installs to docs (#77567)
ansible/ansible: docs/docsite/rst/installation_guide/installation_distros.rst

.. _installing_distros:

Installing Ansible on specific operating systems
================================================

The ``ansible`` package can always be :ref:`installed from PyPI using pip <intro_installation_guide>` on most systems but it is also packaged and maintained by the community for a variety of Linux distributions.

The following instructions will guide you through installing the ``ansible`` package with your preferred distribution's package manager.

.. contents::
  :local:

> Installing Ansible on Fedora or CentOS 
> ----------------------------------------------

Reviewer: nitpick:
```suggestion
Installing Ansible on Fedora or CentOS 
--------------------------------------
```

.. _installing_distros:

Installing Ansible on specific operating systems
================================================

> Follow these instructions to install the Ansible community package on a variety of operating systems.

Reviewer: ```suggestion
The ``ansible`` package can always be :ref:`installed from PyPI using pip <intro_installation_guide>` on most systems but it is also packaged and maintained by the community for a variety of Linux distributions.

The following instructions will guide you through installing the ``ansible`` package with your preferred distribution's package manager.
```

.. contents::
  :local:

Installing Ansible on Fedora or CentOS 
----------------------------------------------

On Fedora:

.. code-block:: bash

    $ sudo dnf install ansible

On CentOS:

.. code-block:: bash

    $ sudo yum install epel-release
    $ sudo yum install ansible

RPMs for currently supported versions of CentOS are also available from `EPEL <https://fedoraproject.org/wiki/EPEL>`_.

> Ansible can manage older CentOS operating systems that contain Python 2.6 or later.

Reviewer: ```suggestion
```

.. _from_apt:

Installing Ansible on Ubuntu
----------------------------

Ubuntu builds are available `in a PPA here <https://launchpad.net/~ansible/+archive/ubuntu/ansible>`_.

To configure the PPA on your system and install Ansible run these commands:

.. code-block:: bash

    $ sudo apt update
    $ sudo apt install software-properties-common
    $ sudo add-apt-repository --yes --update ppa:ansible/ansible
    $ sudo apt install ansible

.. note:: On older Ubuntu distributions, "software-properties-common" is called "python-software-properties". You may want to use ``apt-get`` rather than ``apt`` in older versions. Also, be aware that only newer distributions (that is, 18.04, 18.10, and later) have a ``-u`` or ``--update`` flag. Adjust your script as needed.

> You can also build Debian/Ubuntu packages from the source checkout.

Author: ```suggestion
```

> .. code-block:: bash

Author: ```suggestion
```

>     $ make deb

Author: ```suggestion
```

Installing Ansible on Debian
----------------------------

Debian users can use the same source as the Ubuntu PPA (using the following table).

.. list-table::
  :header-rows: 1

  * - Debian
    -
    - Ubuntu

>   * - Debian 11 (Bullseye)

Reviewer: Based on the existing patterns, I would expect Debian 11 to match Ubuntu jammy (22.04) that should be released anytime soon. Is that correct @Ompragash ?

    - ->
    - Ubuntu 20.04 (Focal)
  * - Debian 10 (Buster)
    - ->
    - Ubuntu 18.04 (Bionic)

.. note::

>     Ansiblereleases are only generated for Ubuntu 18.04 (Bionic) or later releases.

Reviewer: ```suggestion
    Ansible releases are only built for Ubuntu 18.04 (Bionic) or later releases.
```

.. _installing_the_control_node:

> .. _from_yum:

Reviewer 2: I'd use an anchor that matches the page name, for easy links from other pages. If the other anchors are in use, we can leave them there, but following the existing pattern makes for fewer surprises.

Installing Ansible on specific operating systems
================================================

Follow these instructions to install the Ansible community package on a variety of operating systems.

> 

Reviewer 2: Add a local TOC here. See https://docs.ansible.com/ansible/latest/dev_guide/style_guide/#adding-local-tocs.

> Installing Ansible on CentOS or Fedora

Reviewer 2: I'd either change the order in the title, or change the order below to match the order in the title (CentOS, Fedora, CentOS, Fedora . . . or else Fedora, CentOS, Fedora, CentOS).

----------------------------------------------

On Fedora:

.. code-block:: bash

    $ sudo dnf install ansible

On CentOS:

.. code-block:: bash

    $ sudo yum install epel-release
    $ sudo yum install ansible

RPMs for currently supported versions of CentOS are also available from `EPEL <https://fedoraproject.org/wiki/EPEL>`_.

> Ansible can manage older operating systems that contain Python 2.6 or higher.

Reviewer 2: Is this comment specific to CentOS? I'd say "older CentOS versions" if so, or move it if not.

.. _from_apt:

Installing Ansible on Ubuntu
----------------------------

Ubuntu builds are available `in a PPA here <https://launchpad.net/~ansible/+archive/ubuntu/ansible>`_.

To configure the PPA on your machine and install Ansible run these commands:

.. code-block:: bash

    $ sudo apt update
    $ sudo apt install software-properties-common
    $ sudo add-apt-repository --yes --update ppa:ansible/ansible
    $ sudo apt install ansible

.. note:: On older Ubuntu distributions, "software-properties-common" is called "python-software-properties". You may want to use ``apt-get`` instead of ``apt`` in older versions. Also, be aware that only newer distributions (in other words, 18.04, 18.10, and so on) have a ``-u`` or ``--update`` flag, so adjust your script accordingly.

Debian/Ubuntu packages can also be built from the source checkout, run:

.. code-block:: bash

>     $ make deb

Reviewer 2: Maybe add a line to show checking out the code with `git clone`  and then `cd ansible` here? I don't thing `make deb` works from anywhere . . .

Reviewer: We should probably just take this (small) portion out of the page: it's not about building ansible, it's about installing it.

Installing Ansible on Debian
----------------------------

> Debian users may use the same source as the Ubuntu PPA (using the following table).

Reviewer 2: I cannot remember the guidance, does this usage match the rule about translatability for can/may/might?

.. list-table::
  :header-rows: 1

  * - Debian
    -
    - Ubuntu
  * - Debian 11 (Bullseye)
    - ->
    - Ubuntu 20.04 (Focal)
  * - Debian 10 (Buster)
    - ->
    - Ubuntu 18.04 (Bionic)
  * - Debian 9 (Stretch)
    - ->
    - Ubuntu 16.04 (Xenial)
  * - Debian 8 (Jessie)
    - ->
    - Ubuntu 14.04 (Trusty)

.. note::

    As of Ansible 4.0.0, new releases will only be generated for Ubuntu 18.04 (Bionic) or later releases.

Add the following line to ``/etc/apt/sources.list`` or ``/etc/apt/sources.list.d/ansible.list``:

.. code-block:: bash

    deb http://ppa.launchpad.net/ansible/ansible/ubuntu MATCHING_UBUNTU_CODENAME_HERE main

Example for Debian 11 (Bullseye)

.. code-block:: bash

    deb http://ppa.launchpad.net/ansible/ansible/ubuntu focal main

Then run these commands:

.. code-block:: bash

>     $ sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 93C4A3FD7BB9C367

Reviewer 3: apt-key is deprecated

Reviewer: @bcoca my ubuntu/debian is rusty but I looked it up and you're right.

I spun up a debian 11 VM and tested the following commands which appears to be the non-deprecated way:
```
# Download the key and install it to /usr/share/keyrings/ansible-ppa.gpg
curl "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x93C4A3FD7BB9C367" | gpg --dearmor | sudo tee /usr/share/keyrings/ansible-ppa.gpg >/dev/null

# Add the repository with the specified key path
echo "deb [signed-by=/usr/share/keyrings/ansible-ppa.gpg] http://ppa.launchpad.net/ansible/ansible/ubuntu focal main" | sudo tee /etc/apt/sources.list.d/ansible-ppa.list >/dev/null
```
(``>/dev/null`` is to avoid tee re-printing to stdout while elevating privileges just for creating the files)

Reviewer: I can handle this in another PR while simplifying the setup.

    $ sudo apt update
    $ sudo apt install ansible

.. _from_windows:

Installing Ansible on Windows
------------------------------

> See :ref:`windows_faq_ansible`

Reviewer 2: I would add an explicit sentence about "You cannot use a Windows machine as a control node" or "You cannot install Ansible on Windows" or something. That would save most folks a click. The link will provide the curious with more details.
